nano/node/socket.cpp

```diff
@@ -273,25 +268,79 @@ void nano::server_socket::on_connection (std::function<bool (std::shared_ptr<nan
 				new_connection->start_timer (this_l->node.network_params.network.is_dev_network () ? std::chrono::seconds (2) : this_l->node.network_params.node.idle_timeout);
 				this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_accept_success, nano::stat::dir::in);
 				this_l->connections.push_back (new_connection);
+				callback_a (new_connection, ec_a);
+				this_l->on_connection (callback_a);
+				return;
 			}
-			else
+
+			this_l->node.logger.always_log ("Network: Unable to accept connection: ", ec_a.message ());
+			if (this_l->is_temporary_error (ec_a))
 			{
-				this_l->node.logger.try_log ("Network: Unable to accept connection: ", ec_a.message ());
+				this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_accept_failure, nano::stat::dir::in);
+				boost::asio::post (this_l->strand, boost::asio::bind_executor (this_l->strand, [this_l, callback_a] () {
+					this_l->on_connection (callback_a);
+				}));
+				return;
 			}
 
-			// If the callback returns true, keep accepting new connections
-			if (callback_a (new_connection, ec_a))
+			if (this_l->is_system_error (ec_a))
 			{
-				this_l->on_connection (callback_a);
+				boost::this_thread::sleep (boost::posix_time::milliseconds (500));
+				this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_accept_failure, nano::stat::dir::in);
+				boost::asio::post (this_l->strand, boost::asio::bind_executor (this_l->strand, [this_l, callback_a] () {
+					this_l->on_connection (callback_a);
+				}));
+				return;
 			}
-			else
+
+			// Unhandled errors above
+			//EBADF
+			//EFAULT
+			//EINTR
+			//EINVAL
+			//ENOTSOCK
+			//EOPNOTSUPP
+			//EPROTO
+			//EPERM
+			// Check how the listener wants to handle these error
+			if (callback_a (new_connection, ec_a))
 			{
-				this_l->node.logger.try_log ("Network: Stopping to accept connections");
+				this_l->on_connection (callback_a);
+				return;
 			}
+			this_l->node.logger.always_log ("Network: Stopping to accept connections. Node may need restarting.");
 		}));
 	}));
 }
 
+bool nano::server_socket::is_temporary_error (boost::system::error_code const ec_a)
+{
+	switch (ec_a.value ())
+	{
+#if EAGAIN != EWOULDBLOCK
```

Comment:
EAGAIN and EWOULDBLOCK can be different on certain systems but not all. Is this the correct way of handling in the compiler?

Reply:
It triggers a compilation error when they're the same hence this was the best I could find